pkg/controllers/monitoring-stack/controller.go

```diff
@@ -194,9 +199,11 @@ func (r *reconciler) cleanupResources(ctx context.Context, ms *stack.MonitoringS
 		return ctrl.Result{}, nil
 	}
 
-	if err := r.deleteGrafanaDS(ctx, ms); err != nil {
-		logger.V(6).Info("Could not delete GrafanaDataSource", "err", err)
-		return ctrl.Result{}, err
+	if r.enableGrafanaOperator {
```

Comment:
Say we run operator with GO enabled and then re-run it after disabling it, shouldn't we delete existing GrafanaDataSources?